lib/wallet.py

```diff
@@ -2239,7 +2240,17 @@ def sign_transaction(self, tx, password, *, use_cache=False):
         for k in self.get_keystores():
             try:
                 if k.can_sign(tx):
-                    k.sign_transaction(tx, password, use_cache=use_cache)
+                    if ndata:  # used for reusable paycodes
+                        # test if keystore sign_transaction method knows about the `ndata` kwarg
+                        if 'ndata' in inspect.signature(k.sign_transaction, follow_wrapped=True).parameters:
+                            # keystore understands the optional `ndata` kwarg
+                            k.sign_transaction(tx, password, use_cache=use_cache, ndata=ndata)
+                        else:
+                            # keystore does not understand `ndata` (possibly because hw wallet)
+                            raise BaseException("Keystore does not understand ndata parameter.  Possibly wrong wallet type attemping special operation.")
```

Comment:
Why `raise` here?  Also generally `BaseException` is not a good exception to raise.  `RuntimeError` or something is better.. `BaseException` is not supposed to be raised in normal code. 

But why raise?  If you don't raise what happens?  

Note that normally the "contract" is to just continue if `k` cannot sign.

Reply:
Ah ok -- what happens if you don't raise but rather just `continue`?

I guess all the paths that would call this with `ndata` defined catch exceptions and then do the right thing like print an error?

Reply:
>Ah ok -- what happens if you don't raise but rather just continue?

I really don't know.  I believe I would have to see the behavior of that using an actual HW?  I think the way I coded it should be fine but I'm happy to defer to your expertise/decision.

Reply:
I'm just trying to avoid  a situation where you throw and exception and the app gets into a crash reporter. Ideally an error message is printed.  I *think* for the CLI exceptions are caught... but I'm not sure.  Was hoping you could try it.

You could always unconditionally throw (err, `raise` as python calls it) in that code path as a test -- and see what happens. ..

Reply:
As per our discussion, raise BaiseException should be ok.  Although it does produce a traceback on the CLI in addition to the reported error, it should not cause a crash reporter event.